libraries/AP_RangeFinder/AP_RangeFinder_NRA24_CAN.cpp

```diff
@@ -0,0 +1,88 @@
+#include "AP_RangeFinder_NRA24_CAN.h"
+#include <AP_BoardConfig/AP_BoardConfig.h>
+
+#if AP_RANGEFINDER_NRA24_CAN_ENABLED
+
+#include <AP_HAL/AP_HAL.h>
+
+RangeFinder_MultiCAN *AP_RangeFinder_NRA24_CAN::multican_NRA24;
+
+// constructor
+AP_RangeFinder_NRA24_CAN::AP_RangeFinder_NRA24_CAN(RangeFinder::RangeFinder_State &_state, AP_RangeFinder_Params &_params) :
+    AP_RangeFinder_Backend_CAN(_state, _params)
+{
+    if (multican_NRA24 == nullptr) {
+        multican_NRA24 = new RangeFinder_MultiCAN(AP_CAN::Protocol::NanoRadar_NRA24, "NRA24 MultiCAN");
+        if (multican_NRA24 == nullptr) {
+            AP_BoardConfig::allocation_error("Rangefinder_MultiCAN");
+        }
+    }
+
+    {
+        // add to linked list of drivers
+        WITH_SEMAPHORE(multican_NRA24->sem);
```

Comment:
Do we need a nullptr check here?  I see higher up that there's a check of multican_NRA24 so presumably it could still be nullptr here.

Reply:
I don't think we need this check because of the `AP_BoardConfig::allocation_error` call above. But I am not sure.. If we do require it, then we'll have to add this check in a lot of places in code

Reply:
According to PeterB the allocation_error() call does not return.